crates/net/network/src/metrics.rs

```diff
@@ -83,6 +83,8 @@ pub struct NetworkMetrics {
 pub struct SessionManagerMetrics {
     /// Number of successful outgoing dial attempts.
     pub(crate) total_dial_successes: Counter,
+    /// Number of dropped outgoing dial attempts.
+    pub(crate) total_dial_dropped: Counter,
```

Comment:
we need a more appropriate name here because this represents all messages sent to a peer. perhaps.

perhaps `total_outgoing_peer_messages_dropped` ?